### PR TITLE
Break graph on Tensor grad

### PR DESCRIPTION
**Pull request by anijain2305**

### CODE REVIEW

File: torchdynamo/variables/tensor.py [hunk 1]
```diff
@@ -70,6 +70,8 @@ def create(cls, tx, proxy, example_value=None, nnmodule=None, **options):
 
         if isinstance(example_value, torch.Tensor):
             proxy.node.meta["example_value"] = example_value.clone()
+            if example_value.grad is not None:
```
Comment (anijain2305) [inline torchdynamo/variables/tensor.py:73]: There is a hiccup here. PyTorch gives this warning message for the added testcase

~~~
/data/home/anijain/miniconda/envs/pytorch_dev/lib/python3.8/site-packages/torch/_tensor.py:1082: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at  aten/src/ATen/core/TensorBody.h:481.)
~~~

Comment (jansel) [reply torchdynamo/variables/tensor.py:73]: A hrm.  Can we use the `warnings` module to suppress this?

File: torchdynamo/variables/tensor.py [hunk 2]
```diff
@@ -70,6 +70,8 @@ def create(cls, tx, proxy, example_value=None, nnmodule=None, **options):
...
+                proxy.node.meta["example_grad_value"] = example_value.grad.clone()
```
Comment (jansel) [inline torchdynamo/variables/tensor.py:74]: Thoughts on `"example_grad_value"` versus `meta["example_value"].grad` ?

The latter might propagate through some PyTorch ops to other example values, so I'd lean towards that -- but perhaps there are pros of this one I havent thought of?

File: torchdynamo/variables/tensor.py (hunk 1 above)
Comment (anijain2305) [reply torchdynamo/variables/tensor.py:74]: Done

File: torchdynamo/variables/tensor.py (hunk 2 above)
Comment (anijain2305) [reply torchdynamo/variables/tensor.py:75]: I like the `.grad` (latter) version better too. Made the changes.

**Review by jansel — APPROVED** [review]

**Pull request merged**